nifi-nar-bundles/nifi-jmx-bundle/nifi-jmx-processors/src/main/java/org/apache/nifi/processors/jmx/GetJMX.java

```diff
@@ -0,0 +1,406 @@
+        } catch( InstanceNotFoundException|IntrospectionException|ReflectionException e ) {
+            logger.error( "Exception with MBean Server: " + e.getMessage() );
+        } catch( Exception e ) {
+            logger.error( "Exception performing MBean Query: " + e.getMessage() );
+        }
+
+        return metricsSet;
+    }
+
+
+
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+
+        if (queue.size() < context.getProperty(BATCH_SIZE).asInteger()) {
+            final long pollingMillis = context.getProperty(POLLING_INTERVAL).asTimePeriod(TimeUnit.SECONDS) * 1000;
+
+            if ((queueLastUpdated.get() < System.currentTimeMillis() - pollingMillis) && listingLock.tryLock()) {
+                try {
+                    final Set<Record> metrics = performQuery( context );
+
+                    queueLock.lock();
+
+                    try {
+                        metrics.removeAll(inProcess);
+                        metrics.removeAll(recentlyProcessed);
+
+                        queue.clear();
+                        queue.addAll(metrics);
+
+                        queueLastUpdated.set(System.currentTimeMillis());
+                        recentlyProcessed.clear();
+
+                        if (metrics.isEmpty()) {
+                            context.yield();
+                        }
+                    } finally {
+                        queueLock.unlock();
+                    }
+                } finally {
+                    listingLock.unlock();
+                }
+            }
+        }
+
+        final int batchSize = context.getProperty(BATCH_SIZE).asInteger();
+        final List<Record> jmxList = new ArrayList<>(batchSize);
+        queueLock.lock();
+        try {
+            queue.drainTo(jmxList, batchSize);
+            if (jmxList.isEmpty()) {
+                return;
+            } else {
+                inProcess.addAll(jmxList);
+            }
+        } finally {
+            queueLock.unlock();
+        }
+
+        final ListIterator<Record> itr = jmxList.listIterator();
+
+        FlowFile flowFile = null;
+        try {
+            while (itr.hasNext()) {
+                final Record record = itr.next();
+
+                final Map<String, String> attributes = new HashMap<>();
+                final Timestamp timestamp = new Timestamp(System.currentTimeMillis());
+
+                attributes.put(HOSTNAME_ATTRIBUTE, context.getProperty(HOSTNAME).getValue());
+                attributes.put(PORT_ATTRIBUTE, context.getProperty(PORT).getValue());
+                attributes.put(TIMESTAMP_ATTRIBUTE, timestamp.toString());
+
+                flowFile = session.create();
+                flowFile = session.putAllAttributes(flowFile, attributes);
+                logger.info("ATTRIBUTES: " + context.getProperty(HOSTNAME).getValue() + " " +
```

Comment:
I'd consider debug level here, otherwise, could generate quite a lot of logs based on how the processor is configured.